cognite/client/data_classes/simulators/simulators.py

```diff
@@ -0,0 +1,1645 @@
+        return SimulatorRoutineRevisionWrite(
+            external_id=self.external_id,
+            simulator_external_id=self.simulator_external_id,
+            routine_external_id=self.routine_external_id,
+            simulator_integration_external_id=self.simulator_integration_external_id,
+            model_external_id=self.model_external_id,
+            data_set_id=self.data_set_id,
+            configuration=self.configuration,
+            script=self.script,
+        )
+
+
+class SimulatorModelCore(WriteableCogniteResource["SimulatorModelWrite"], ABC):
+    """
+    The simulator model resource represents an asset modeled in a simulator.
+    This asset could range from a pump or well to a complete processing facility or refinery.
+    The simulator model is the root of its associated revisions, routines, runs, and results.
+    The dataset assigned to a model is inherited by its children. Deleting a model also deletes all its children, thereby
+    maintaining the integrity and hierarchy of the simulation data.
+
+    Simulator model revisions track changes and updates to a simulator model over time.
+    Each revision ensures that modifications to models are traceable and allows users to understand the evolution of a given model.
+
+    This is the read/response format of a simulator model.
+
+    Args:
+        external_id (str | None): External id of the simulator model
+        simulator_external_id (str | None): External id of the associated simulator
+        data_set_id (int | None): The id of the dataset associated with the simulator model
+        name (str | None): The name of the simulator model
+        type (str | None): The type key of the simulator model
+        description (str | None): The description of the simulator model
+    """
+
+    def __init__(
+        self,
+        external_id: str | None = None,
+        simulator_external_id: str | None = None,
+        data_set_id: int | None = None,
+        name: str | None = None,
+        type: str | None = None,
+        description: str | None = None,
+    ) -> None:
+        self.external_id = external_id
+        self.simulator_external_id = simulator_external_id
+        self.data_set_id = data_set_id
+        self.name = name
+        self.type = type
+        self.description = description
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        instance = super()._load(resource, cognite_client)
+        return instance
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return super().dump(camel_case=camel_case)
+
+
+class SimulatorModelWrite(SimulatorModelCore):
+    def __init__(
+        self,
+        external_id: str | None = None,
+        simulator_external_id: str | None = None,
+        data_set_id: int | None = None,
+        name: str | None = None,
+        type: str | None = None,
+        description: str | None = None,
```

Comment:
If I'm reading the API spec right, all of these should be required and only description optional?